common/namespace/registry.go

```diff
@@ -161,6 +167,11 @@ type (
 		cacheNameToID        cache.Cache
 		cacheByID            cache.Cache
 		stateChangeCallbacks map[any]StateChangeCallbackFn
+
+		// readthroughCache stores namespaces that missed the above caches
+		// AND was not found when reading through to the persistence layer
+		readthroughLock  sync.RWMutex
+		readthroughCache cache.Cache
```

Comment:
this should be named readthroughErrorsCache? from the name it sounds like it's storing namespaces, not errors, and it was confusing on the first read-through